packages/nuxt/src/runtime/hooks/wrapMiddlewareHandler.ts

```diff
@@ -0,0 +1,122 @@
+import {
+  type RequestEventData,
+  type SpanAttributes,
+  captureException,
+  debug,
+  flushIfServerless,
+  getClient,
+  getDefaultIsolationScope,
+  getIsolationScope,
+  httpHeadersToSpanAttributes,
+  httpRequestToRequestData,
+  SEMANTIC_ATTRIBUTE_SENTRY_OP,
+  SEMANTIC_ATTRIBUTE_SENTRY_ORIGIN,
+  SEMANTIC_ATTRIBUTE_SENTRY_SOURCE,
+  SPAN_STATUS_ERROR,
+  SPAN_STATUS_OK,
+  startSpan,
+  withIsolationScope,
+} from '@sentry/core';
+import type { EventHandler, EventHandlerRequest, H3Event } from 'h3';
+
+/**
+ * Wraps a middleware handler with Sentry instrumentation.
+ *
+ * @param handler The middleware handler.
+ * @param fileName The name of the middleware file.
+ */
+export function wrapMiddlewareHandler(handler: EventHandler, fileName: string) {
+  return async (event: H3Event<EventHandlerRequest>) => {
+    debug.log(`Sentry middleware: ${fileName} handling ${event.path}`);
+
+    const isolationScope = getIsolationScope();
+    const newIsolationScope = isolationScope === getDefaultIsolationScope() ? isolationScope.clone() : isolationScope;
+    const normalizedRequest = createNormalizedRequestData(event);
+    newIsolationScope.setSDKProcessingMetadata({
+      normalizedRequest,
+    });
+
+    const attributes = getSpanAttributes(event, fileName);
+
+    return withIsolationScope(newIsolationScope, async () => {
+      return startSpan(
+        {
+          name: fileName,
+          attributes,
+        },
+        async span => {
+          try {
+            const result = await handler(event);
+            span.setStatus({ code: SPAN_STATUS_OK });
+
+            return result;
+          } catch (error) {
+            span.setStatus({ code: SPAN_STATUS_ERROR, message: 'internal_error' });
```

Comment:
this is also not strictly necessary, as long as we re-throw the error the span status is automatically set to errored.